manager/controllers/utils.go

```diff
@@ -324,9 +324,11 @@ func (t *Tag) ApplyTag(tx *gorm.DB) *gorm.DB {
 }
 
 // Filter systems by tags,
-func ApplyTagsFilter(c *gin.Context, tx *gorm.DB, systemIDExpr string) (*gorm.DB, bool, error) {
+// nolint: funlen
+func ApplyTagsFilter(c *gin.Context, tx *gorm.DB, systemIDExpr string) (*gorm.DB, bool, map[string]FilterData, error) {
```

Comment:
This function is now too long and does too many things. We could separate filters applying and parsing into two separate functions so each will be shorter, better readable and testable.

I think we could firstly parse filters data`func ParseTagsFilters(c *gin.Context) map[string]FilterData`, then we can use this result for sql query update `func ApplytagsFilter(tx *gorm.DB, map[string]FilterData) *gorm.DB` or so.